File: problems/Python/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        msum = nums[0]
        csum = nums[0]
        
        for i in range(1, len(nums)):
            n = nums[i]
            csum = max(n, csum + n)
            msum = max(msum, csum)
            
        return msum

File: problems/C++/53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int msum = nums[0];
        int csum = nums[0];

        for (int i = 1; i < nums.size(); ++i) {
            int n = nums[i];
            csum = max(n, csum + n);
            msum = max(msum, csum);
        }
        
        return msum;
    }
};